lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
+          if (!options.reconnect) {
+            log("connection closed; reconnect not enabled");
+            reconnectUI.showDisconnected();
```

Comment:
I can't find any usages of `pc.allowReconnect` in the project (other than setting the value), so I didn't worry about it here.